crates/autopilot/src/domain/settlement/solution/mod.rs

```diff
@@ -194,6 +200,16 @@ pub mod error {
     }
 }
 
+/// Fee per trade in a solution. Contains breakdown of protocol fees and total
+/// fee.
+#[derive(Debug, Clone)]
+pub struct OrderFee {
+    /// Gas fee + protocol fees. Total fee taken for the execution of the order.
+    pub total: eth::SellTokenAmount,
+    /// Breakdown of protocol fees.
+    pub protocol: Vec<eth::Asset>,
```

Comment:
Should we also store the fee policy enum for each `protocol` fee?

Reply:
Otherwise we should clearly document that the lost of fees in `order_fees` aligns with the list of fee policies (but to me this is implicit knowledge which is easy to break unintentionally in a future refactoring, so I'd prefer being explicit).

Reply:
Fee policies don't have the ID, so if we, for example, have two volume based policies for an order, we still need to have an implicit ordering, even if we attach the fee policies here.

Reply:
I think the main thing people want to know is that when there is a fee of $10 and $50, which fee was for which policy (ie volume vs. surplus). Right now this is not easy to do (it's not documented and requires zipping with an unrelated field) and sure you need the ordering in order to re-compute the values accurately in order, but the point of this PR is precisely not having to do that in the client.

Reply:
What about the database order_execution table? Would you expect the fee policy saved there as well?